### PR TITLE
RecentReposSettings: Streamline naming

### PR DESCRIPTION
**Pull request by pmiossec**
between "Top", "Recent" and "Anchored" repositories

Also:
* inverted the 2 top settings to be in the same order than list it is referring to.
* used similar label as 2 settings are referring the the same behavior for top and recent lists..

Discussed [here](https://github.com/gitextensions/gitextensions/pull/11779#issuecomment-2164942593)

### Before

image

### After

image

## Test methodology 

- Manual

## Test environment(s) 

- GIT 
- Windows 

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by mstv — APPROVED** [review]
LGTM, have not run

File: src/app/GitCommands/Settings/AppSettings.cs [hunk 1]
```diff
@@ -1678,7 +1678,7 @@ public static ShorteningRecentRepoPathStrategy ShorteningRecentRepoPathStrategy
             set => SetEnum("ShorteningRecentRepoPathStrategy", value);
         }
 
-        public static int MaxPinnedRepositories
+        public static int MaxTopRepositories
         {
             get => GetInt("MaxMostRecentRepositories", 0);
```
Comment (gerhardol) [inline src/app/GitCommands/Settings/AppSettings.cs:1683]: Add comment why it differ?
(I did not do that when I was asked to rename.)

**Review by gerhardol — APPROVED** [review]

**Review by RussKie — APPROVED** [review]

File: src/app/GitCommands/Settings/AppSettings.cs (hunk 1 above)
Comment (pmiossec) [reply src/app/GitCommands/Settings/AppSettings.cs:1684]: Done (with some little comment refactorings)

Comment (gerhardol) [reply src/app/GitCommands/Settings/AppSettings.cs:1684]: Thanks for cleaning up

**Review by gerhardol — APPROVED** [review]

**Pull request merged**